mantle/platform/qemu.go

```diff
@@ -900,6 +900,43 @@ func (builder *QemuBuilder) SerialPipe() (*os.File, error) {
 	return r, nil
 }
 
+// VirtioJournal configures the OS and VM to stream the systemd journal
+// (post-switchroot) over a virtio-serial channel.  The first return value
+// is an Ignition fragment that should be included in the target config,
+// and the file stream will be newline-separated JSON.  The optional
+// queryArguments filters the stream - see `man journalctl` for more information.
+func (builder *QemuBuilder) VirtioJournal(queryArguments string) (*v3types.Config, *os.File, error) {
+	stream, err := builder.VirtioChannelRead("mantlejournal")
+	if err != nil {
+		return nil, nil, err
+	}
+	var streamJournalUnit = fmt.Sprintf(`[Unit]
+	Requires=dev-virtio\\x2dports-mantlejournal.device
```

Comment:
This should also have e.g. `After=systemd-journald.socket`, no?

Reply:
That's implicitly part of `DefaultDependencies=yes` since all socket units are done by `basic.target`.